lib/Dialect/Secret/IR/SecretOps.cpp

```diff
@@ -94,6 +112,59 @@ ParseResult GenericOp::parse(OpAsmParser &parser, OperationState &result) {
   return success();
 }
 
+LogicalResult GenericOp::verify() {
+  Block *body = getBody();
+
+  // Verify that the operands of the body's basic block are the non-secret
+  // analogues of the generic's operands.
+  for (BlockArgument arg : body->getArguments()) {
+    auto operand = getOperands()[arg.getArgNumber()];
+    auto operandType = dyn_cast<SecretType>(operand.getType());
+    if (!operandType && arg.getType() != operand.getType()) {
+      return emitOpError()
+             << "Type mismatch between block argument " << arg.getArgNumber()
+             << " of type " << arg.getType()
+             << " and generic operand argument of type " << operand.getType()
+             << ". If the operand is not secret, it must be the same type as "
+                "the block argument";
+    }
+
+    if (operandType && arg.getType() != operandType.getValueType()) {
+      return emitOpError() << "Type mismatch between block argument "
+                           << arg.getArgNumber() << " of type " << arg.getType()
+                           << " and generic operand argument of type "
+                           << operand.getType()
```

Comment:
Im guessing Operand.getType() will print out the ValueType within Secret.secret.

Im slightly confused by the error message on this one being similar to the one above for non-secret oeprand. Perhaps update the message to "type mismatch between block argument <> of type <> and **ValueType** of generic operand argument of type <>"

Reply:
Clarified